qiskit/opflow/list_ops/list_op.py

```diff
@@ -72,12 +72,19 @@ def __init__(self,
             identity - it accepts the list of values, and returns them in a list.
         """
         super().__init__()
-        self._oplist = list(oplist)
+        self._oplist = self._get_op_list(oplist)
         self._combo_fn = combo_fn
         self._coeff = coeff
         self._abelian = abelian
         self._grad_combo_fn = grad_combo_fn
 
+    def _get_op_list(self, oplist):
```

Comment:
Could we rename this so it's clear the main purpose is verification? Maybe `_check_input_types` or so? E.g.
```
self._check_input_types(oplist)

super().__init__()
self._oplist = list(oplist)
```

Reply:
Yes, that would be better.